atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTimestampBoundStore.java

```diff
@@ -120,67 +129,87 @@ public synchronized void storeUpperLimit(final long limit) {
         clientPool.runWithRetry(new FunctionCheckedException<Client, Void, RuntimeException>() {
             @Override
             public Void apply(Client client) {
-                cas(client, currentLimit, limit);
+                cas(client, makeNewColumn(currentLimit), currentLimit, limit);
                 return null;
             }
         });
     }
 
-    private void cas(Client client, Long oldVal, long newVal) {
-        final CASResult result;
+    private void cas(Client client, Column oldColumn, Long oldVal, long newVal) {
+        CASResult result = updateTimestampInDb(client, oldColumn, oldVal, newVal);
+        if (!result.isSuccess()) {
+            if (result.getCurrent_values().size() == 0) {
+                log.error("[CAS] Error trying to set timestamp limit to {}: there is no limit stored in DB.", newVal);
+                throw new Error("Error trying to set timestamp limit: there is no limit stored in DB.");
```

Comment:
The old behaviour was actually to throw a MultipleRunningTimestampServiceError, so maybe we should just throw that? Or make a new error or even extend MultipleRunningTimestampServiceError?